@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1302,6 +1309,79 @@ export class TextElementsRenderer {
+                // Get the screen points that define the label's segments and create a path with
+                // them.
+                screenPoints = [];
+                let minX = Number.MAX_SAFE_INTEGER;
+                let maxX = Number.MIN_SAFE_INTEGER;
+                let minY = Number.MAX_SAFE_INTEGER;
+                let maxY = Number.MIN_SAFE_INTEGER;
+                const tileCenterX = textElement.tileCenterX!;
+                const tileCenterY = textElement.tileCenterY!;
+                for (const pt of textElement.path!) {
+                    tempPosition.set(pt.x + tileCenterX, pt.y + tileCenterY, 0);
+                    const screenPoint = this.m_screenProjector.project(tempPosition);
+                    if (screenPoint === undefined) {
+                        continue;
+                    }
+                    screenPoints.push(screenPoint);
+
+                    if (screenPoint.x < minX) {
+                        minX = screenPoint.x;
+                    }
+                    if (screenPoint.x > maxX) {
+                        maxX = screenPoint.x;
+                    }
+                    if (screenPoint.y < minY) {
+                        minY = screenPoint.y;
+                    }
+                    if (screenPoint.y > maxY) {
+                        maxY = screenPoint.y;
+                    }
+
+                    if (indexOfFirstVisibleScreenPoint < 0) {
+                        const firstIndex = screenPoints.findIndex(p2 => {
+                            if (this.m_screenCollisions.screenBounds.contains(p2.x, p2.y)) {
+                                return true;
+                            }
+                            return false;
+                        });
+
+                        if (firstIndex >= 0) {
+                            indexOfFirstVisibleScreenPoint = firstIndex;
+                        }
+                    }
+                }
+
+                // TODO: (HARP-3515)
+                //      The rendering of a path label that contains just a single point that is not
+                //      visible is impossible, which is problematic with long paths.
+                //      Fix: Skip/clip the invisible points at beginning and end of the path to get
+                //      the visible part of the path.
+
+                // If not a single point is visible, skip the path
+                if (indexOfFirstVisibleScreenPoint === -1) {
+                    numNotVisible++;
+                    continue;
+                }
+
+                // Check/guess if the screen box can hold a string of that length. It is important
+                // to guess that value without measuring the font first to save time.
+                const minScreenSpace = textElement.text.length * MIN_PATH_SCREEN_LENGTH;
+                const minScreenSpaceSqr = minScreenSpace * minScreenSpace;
+                if (
+                    (maxX - minX) * (maxX - minX) < minScreenSpaceSqr &&
```

Comment:
minor this subtraction is repeated twice, is javascript efficient enough to optimize this out?

Reply:
Hard to say, calling Math.sqr() would probably be slower, using a temporary may be faster... Probably depends on the compiler.

Reply:
I don't mean call Math.sqr, that is for sure slower, I just mean put "(maxX - minX)" in some variable, because it is computed twice, but anyway, minor.

Reply:
Keeping current code.